src/driver/sqlserver/SqlServerQueryRunner.ts

```diff
@@ -1288,6 +1289,15 @@ export class SqlServerQueryRunner extends BaseQueryRunner implements QueryRunner
      */
     async createForeignKey(tableOrName: Table|string, foreignKey: TableForeignKey): Promise<void> {
         const table = tableOrName instanceof Table ? tableOrName : await this.getCachedTable(tableOrName);
+        
+        let metadata: EntityMetadata | undefined = undefined;
+        try {
+            metadata = this.connection.getMetadata(table.name);
+        } catch (err) { } // Ignore error and continue
```

Comment:
`try ...catch` can be replaced with `hasMetadata(table.name)` condition

Reply:
Thanks for the catch, fixed.